packages/@aws-cdk/aws-batch/lib/managed-compute-environment.ts

```diff
@@ -0,0 +1,1222 @@
+   * The Launch Template that this Compute Environment
+   * will use to provision EC2 Instances.
+   *
+   * *Note*: if `securityGroups` is specified on both your
+   * launch template and this Compute Environment, **the
+   * `securityGroup`s on the Compute Environment override the
+   * ones on the launch template.**
+   *
+   * @default - no launch template
+   */
+  readonly launchTemplate?: ec2.ILaunchTemplate;
+
+  /**
+   * The minimum vCPUs that an environment should maintain,
+   * even if the compute environment is DISABLED.
+   *
+   * @default 0
+   */
+  readonly minvCpus?: number;
+
+  /**
+   * The EC2 placement group to associate with your compute resources.
+   * If you intend to submit multi-node parallel jobs to this Compute Environment,
+   * you should consider creating a cluster placement group and associate it with your compute resources.
+   * This keeps your multi-node parallel job on a logical grouping of instances
+   * within a single Availability Zone with high network flow potential.
+   *
+   * @see: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html
+   *
+   * @default - no placement group
+   */
+  readonly placementGroup?: ec2.IPlacementGroup;
+}
+
+/**
+ * A ManagedComputeEnvironment that uses ECS orchestration on EC2 instances.
+ *
+ * @resource AWS::Batch::ComputeEnvironment
+ */
+export class ManagedEc2EksComputeEnvironment extends ManagedComputeEnvironmentBase implements IManagedEc2EksComputeEnvironment {
+  public readonly kubernetesNamespace?: string;
+  public readonly eksCluster: eks.ICluster;
+
+  public readonly computeEnvironmentName: string;
+  public readonly computeEnvironmentArn: string;
+
+  public readonly images?: EksMachineImage[];
+  public readonly allocationStrategy?: AllocationStrategy;
+  public readonly spotBidPercentage?: number;
+  public readonly spotFleetRole?: iam.IRole | undefined;
+  public readonly instanceTypes: ec2.InstanceType[];
+  public readonly instanceClasses: ec2.InstanceClass[];
+  public readonly instanceRole?: iam.IRole;
+  public readonly launchTemplate?: ec2.ILaunchTemplate;
+  public readonly minvCpus?: number;
+  public readonly placementGroup?: ec2.IPlacementGroup;
+
+  private readonly instanceProfile: iam.CfnInstanceProfile;
+
+  constructor(scope: Construct, id: string, props: ManagedEc2EksComputeEnvironmentProps) {
+    super(scope, id, props);
+
+    this.kubernetesNamespace = props.kubernetesNamespace;
+    this.eksCluster = props.eksCluster;
+
+    this.images = props.images;
+    this.allocationStrategy = determineAllocationStrategy(id, props.allocationStrategy, this.spot);
+    if (this.allocationStrategy === AllocationStrategy.BEST_FIT) {
+      throw new Error(`ManagedEc2EksComputeEnvironment '${id}' uses invalid allocation strategy 'AllocationStrategy.BEST_FIT'`);
+    }
+    this.spotBidPercentage = props.spotBidPercentage;
+    this.spotFleetRole = props.spotFleetRole ?? (this.spot ? createSpotFleetRole(this) : undefined);
```

Comment:
EKS CE does not support BEST_FIT SPOT, therefore spotFleetRole is never needed in for any EKS CE allocation type.